server/src/main/java/org/elasticsearch/plugins/HealthIndicatorPlugin.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.plugins;
+
+import org.elasticsearch.health.HealthIndicatorService;
+
+import java.util.stream.Stream;
+
+/**
+ * An additional extension point for {@link Plugin}s that extends Elasticsearch's health indicators functionality.
+ */
+public interface HealthIndicatorPlugin {
+
+    Stream<HealthIndicatorService> getHealthIndicatorServices();
```

Comment:
I think it would be better to ask a plugin to return a `Collection` instead of `Stream`. Otherwise, we can end up in situations where `getHealthIndicatorServices` can be called only once. A plugin which returns `Stream.of`, for example. I think it would be better to be explicit ask for a `Collection` so any consumers can call `getHealthIndicatorServices` safely. 

See _Effective Java, Item 47 : Prefer Collection to Stream as a return type_

Reply:
:+1: